keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -35,38 +42,51 @@ class MemoryCloudDriverCache(
   private val securityGroupSummariesByIdOrName = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, MINUTES)
-    .build<String, SecurityGroupSummary>()
+    .buildAsync<String, SecurityGroupSummary>()
 
   private val networks = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Network>()
+    .buildAsync<String, Network>()
 
   private val availabilityZones = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Set<String>>()
+    .buildAsync<String, Set<String>>()
 
   private val credentials = Caffeine.newBuilder()
     .maximumSize(100)
     .expireAfterWrite(1, HOURS)
-    .build<String, Credential>()
+    .buildAsync<String, Credential>()
 
   private val subnetsById = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Subnet>()
+    .buildAsync<String, Subnet>()
 
   private val subnetsByPurpose = Caffeine.newBuilder()
     .maximumSize(1000)
     .expireAfterWrite(1, HOURS)
-    .build<String, Subnet>()
+    .buildAsync<String, Subnet>()
 
   override fun credentialBy(name: String): Credential =
     credentials.getOrNotFound(name, "Credentials with name $name not found") {
       cloudDriver.getCredential(name, DEFAULT_SERVICE_ACCOUNT)
     }
 
+  /**
+   * Convert a value to a [CompletableFuture] so it can be passed to [AsyncCache.put]
+   */
+  private fun <T> futurify(x: T?): CompletableFuture<T?> = CoroutineScope(EmptyCoroutineContext).future { x }
```

Comment:
Because I don't have a deep understanding of kotlin coroutines, I'm not sure if I'm using right coroutine scope here.

Reply:
Frankly, I don't think any of us are experts in coroutines. 🙂 

Could you please elaborate on what led you to pick `EmptyCoroutineContext` in this case?

Reply:
The [future] method is defined on a coroutine scope, and TBH I couldn't think of a better option for a coroutine scope to use here. This code just wraps a constant value inside of a future (i.e., all it does is `return x` where x is a known value). My reasoning was to use the simplest possible thing as a wrapper, and `EmptyCoroutineContext` sounded simplest. 

I could also avoid coroutines entirely and use the [supplyAsync] method:

```
private fun <T> futurify(x: T?): CompletableFuture<T?> = applyAsync {x}
```

But when I read the description of that method, it sounds like it will execute the lambda inside of a threadpool, which is OK, but seems wasteful to do a thread context switch instead of a coroutine context switch.

[future]: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-jdk8/kotlinx.coroutines.future/kotlinx.coroutines.-coroutine-scope/future.html

[supplyAsync]: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#supplyAsync-java.util.function.Supplier-

Reply:
what about `CompletableFuture.completedFuture(x)`?

Reply:
if this is for `AsyncCache.put(key, future)` per comment, then you can use the `synchronous()` view's `put(key, value)`, which does `put(key, completedFuture(value))`

Reply:
> if this is for `AsyncCache.put(key, future)` per comment, then you can use the `synchronous()` view's `put(key, value)`, which does `put(key, completedFuture(value))`

Yup, this is what it's for. I switched to using `synchronous().put(...)` in 83982e61.